repos/fdbt-site/src/pages/api/defineCaps.ts

```diff
@@ -0,0 +1,68 @@
+import { NextApiResponse } from 'next';
+import { CAPS_DEFINITION_ATTRIBUTE } from '../../constants/attributes';
+import { getCapByNocAndId } from '../../data/auroradb';
+import { getAndValidateNoc, redirectTo, redirectToError } from '../../utils/apiUtils/index';
+import { CapInfo, CapsDefinitionWithErrors, NextApiRequestWithSession } from 'src/interfaces';
+import { updateSessionAttribute } from '../../../src/utils/sessions';
+
+const getCapContent = async (cap: number, noc: string): Promise<CapInfo | undefined> => {
+    const result = await getCapByNocAndId(noc, cap);
+
+    if (!result) {
+        throw new Error('No premade caps saved under the provided name');
+    }
+
+    return result;
+};
+
+export default async (req: NextApiRequestWithSession, res: NextApiResponse): Promise<void> => {
+    try {
+        const { capChoice, cap } = req.body;
+
+        if (!capChoice) {
+            const capDefinitionWithErrors: CapsDefinitionWithErrors = {
+                id: cap,
+                capChoice,
+                fullCaps: [],
+                errors: [{ errorMessage: 'Choose one of the options below', id: 'no-caps' }],
+            };
+
+            updateSessionAttribute(req, CAPS_DEFINITION_ATTRIBUTE, capDefinitionWithErrors);
+            redirectTo(res, '/selectCaps');
+            return;
+        }
+
```

Comment:
I think the first thing we need to do after the check to see if capChoice is defined, is to see if they selected no. 

If they selected no, we can just redirect them onwards and not do anything else.

And then underneath that, we can have all the code for them selecting yes.